netket/operator/_kinetic.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2021 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional, Callable, Union, List
+
+from netket.utils.types import DType, PyTree, Array
+
+from netket.hilbert import AbstractHilbert
+from netket.operator import ContinousOperator
+
+import jax
+import jax.numpy as jnp
+
+
+class KineticEnergy(ContinousOperator):
+    r"""This is the kinetic energy operator (hbar = 1). The local value is given by:
+    :math:`E_{kin} = -1/2 ( \sum_i \frac{1}{m_i} (\log(\psi))'^2 + (\log(\psi))'' )`
+    """
+
+    def __init__(
+        self,
+        hilbert: AbstractHilbert,
+        mass: Union[float, List[float]],
+        dtype: Optional[DType] = float,
+    ):
+        r"""Args:
+        hilbert: The underlying Hilbert space on which the operator is defined
+        mass: float if all masses are the same, list indicating the mass of each particle otherwise
+        dtype: Data type of the matrix elements. Defaults to `np.float64`
+        """
+
+        self._dtype = dtype
+        super().__init__(hilbert, self._dtype)
+
+        self._mass = jnp.asarray(mass, dtype=self._dtype)
```

Comment:
```suggestion
        dtype: Optional[DType] = None,
    ):
        r"""Args:
        hilbert: The underlying Hilbert space on which the operator is defined
        mass: float if all masses are the same, list indicating the mass of each particle otherwise
        dtype: Data type of the matrix elements. Defaults to `np.float64`
        """
        
        self._mass = jnp.asarray(mass, dtype=dtype)
     
        super().__init__(hilbert, self._mass.dtype)
```